app/grants/management/commands/subminer.py

```diff
@@ -99,6 +102,49 @@ def process_subscription(subscription, live):
                     subscription.save()
 
 
+def listen_for_tx(grant, subscription, tx, network, tx_type):
+    if grant:
+        logger.info("  - grant %d %s", grant.pk, tx_type)
+    elif subscription:
+        logger.info("  - sub %d %s", subscription.pk, tx_type)
+
+    status = 'failure'
+    error = "none"
+    logger.info("   -- *waiting for confirmation* ")
+    while not has_tx_mined(tx, network):
+        time.sleep(SLEEP_TIME)
+        logger.info(f"   -- *waiting {SLEEP_TIME} seconds*")
+    status, __, tx_contract_address = get_tx_status(tx, network, timezone.now())
+    if status != 'success':
+        error = f"tx status from RPC is {status} not success, txid: {tx}"
+
+    logger.info("   -- *mined* (status: %s / error: %s) ", status, error)
+    was_success = status == 'success'
+    if not was_success:
+            logger.warning('tx processing failed')
+            # TODO: Add failed_transaction email
```

Comment:
is this still TODO?

Reply:
Yes, when working on this a few weeks ago there was a weird issue with the email urls or something. We were trying to get it shipped before I left and so you suggested I leave it as a TODO for now to the PR out. That didnt happen though. Do you want me to implement the email? @owocki

Reply:
Isn't this the email already available in this PR ? `transaction_failed.html` ?

Reply:
@thelostone-mc It is available, but it is not rendering in the admin view. I didnt complete it because it was taking more time than desired and was less important than getting the full PR merged. I can take the time to implement now if desired.

Reply:
This has been implemented